eventstore/src/test/groovy/com/github/yafna/events/pipelines/EventPipelineSpec.groovy

```diff
@@ -0,0 +1,61 @@
+package com.github.yafna.events.pipelines
+
+import com.github.yafna.events.Event
+import com.github.yafna.events.dispatcher.GsonEventDispatcher
+import com.github.yafna.events.handlers.count.RabbitCountingHandler
+import com.github.yafna.events.handlers.count.Rabbits
+import com.github.yafna.events.rabbits.Rabbit
+import com.github.yafna.events.rabbits.RabbitAdded
+import com.github.yafna.events.store.file.GsonFileEventStore
+import com.google.gson.Gson
+import spock.lang.Specification
+
+import java.time.Clock
+import java.util.concurrent.ForkJoinPool
+import java.util.concurrent.TimeUnit
+import java.util.stream.Collectors
+
+class EventPipelineSpec extends Specification {
+
+    ForkJoinPool executor = new ForkJoinPool()
+    Clock clock = Clock.systemUTC()
+    GsonFileEventStore store = new GsonFileEventStore(clock, File.createTempDir())
+
+    def 'given [rabbit.added] event persisted should call EventHandler for it'() {
+        given:
+            def handler = new RabbitCountingHandler()
+            EventPipeline<Rabbit, RabbitAdded> subj = new EventPipeline(
+                    new GsonEventDispatcher(store, executor), RabbitAdded.class, handler, "rabbit-counter", clock, executor 
+            )
```

Comment:
The handler is created in this test and the handlers is test code. Such check would be excessive